packages/@aws-cdk/aws-appmesh/lib/tls-certificate.ts

```diff
@@ -0,0 +1,173 @@
+import * as acm from '@aws-cdk/aws-certificatemanager';
+import * as cdk from '@aws-cdk/core';
+import { CfnVirtualNode } from './appmesh.generated';
+
+/**
+ * Enum of supported TLS modes
+ */
+export enum TlsMode {
+  /**
+   * Only accept encrypted traffic
+   */
+  STRICT = 'STRICT',
+
+  /**
+   * Accept encrypted and plaintext traffic.
+   */
+  PERMISSIVE = 'PERMISSIVE',
+
+  /**
+   * TLS is disabled, only accept plaintext traffic.
+   */
+  DISABLED = 'DISABLED',
+}
+
+/**
+ * A wrapper for the tls config returned by {@link TlsCertificate.bind}
+ */
+export interface TlsCertificateConfig {
+  /**
+   * The CFN shape for a listener TLS certificate
+   */
+  readonly tlsCertificate: CfnVirtualNode.ListenerTlsCertificateProperty,
+
+  /**
+   * The TLS mode.
+   */
+  readonly tlsMode: TlsMode;
+}
+
+/**
+ * ACM Certificate Properties
+ */
+export interface AcmCertificateOptions {
+  /**
+   * The TLS mode.
+   */
+  readonly tlsMode: TlsMode;
```

Comment:
Why did you change this to be required? Can't we establish a sensible default here, maybe `PERMISSIVE`?

Reply:
For listener TLS, mode is a required field ([API reference](https://docs.aws.amazon.com/app-mesh/latest/APIReference/API_ListenerTls.html)) and setting a default here would be taking a security stance.

I initially wrote the TLS shape as a flattened version of the one in AppMesh's spec. However I just noticed that [Client TLS validation was implemented](https://github.com/aws/aws-cdk/pull/11563) without flattening the shape. I'm considering refactoring this PR to have parity within our L2 CDK implementation.

Reply:
I think we should leave it a required field. It is a security feature and we want to ensure customers are setting it to the level they need. If we were to default, I would want us to default to `STRICT` so customers do not think they are enforcing TLS by specifying a certificate when in fact plaintext connections are still allowed

Reply:
OK, I can buy that 🙂.